test/e2e/network/dns.go

```diff
@@ -88,6 +90,38 @@ var _ = common.SIGDescribe("DNS", func() {
 		validateDNSResults(ctx, f, pod, append(wheezyFileNames, jessieFileNames...))
 	})
 
+	/*
+		Release: v1.30
+		Testname: DNS, cluster
+		Description: When a Pod is created, the container hostname should match with spec hostname after setting the hostNetwork to true
+	*/
+	framework.ConformanceIt("should resolve the hostname and hostNetwork for a Pod", func(ctx context.Context) {
+		ginkgo.By("Creating a pod by setting hostname")
+		execPod := e2epod.CreateExecPodOrFail(ctx, f.ClientSet, f.Namespace.Name, dnsTestPodHostName, nil)
+		cmd := "hostname"
+		execPod.Spec.Hostname = dnsTestPodHostName
+		framework.Logf("Setting up hostname to: %v", execPod.Spec.Hostname)
+		stdout, err := e2eoutput.RunHostCmd(execPod.Namespace, execPod.Name, cmd)
+		if err != nil {
+			framework.Logf("Error running hostname command: %v", err)
+		}
+		trimmed := strings.TrimSpace(stdout)
+		gomega.Expect(execPod.Spec.Hostname).NotTo(gomega.Equal(trimmed))
+		framework.Logf("The spec.hostname is not same as container hostname, expected to contain: %s, got: %s", execPod.Spec.Hostname, stdout)
+
+		ginkgo.By("Updating the pod spec.hostNetwork to true")
+		execPod.Spec.HostNetwork = true
```

Comment:
this is not doing what you think is doing, you can not mutate a pod from network to hostnetwork, that is not possible

Reply:
Should I not update the HostNetwork.
Is there any other way to update that or it is not possible to update?

Reply:
Yash you need to configure `hostNetwork` true before creating the pod.
Something similar to this - https://github.com/kubernetes/kubernetes/blob/master/test/e2e/network/conntrack.go#L301-L308